File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.cpp
#include <string>
#include <vector>
#include <sstream> 
using namespace std;

string solution(string polynomial) {
    string answer = "";
    stringstream ss(polynomial);
    string temp = "";
    vector<string> data;
    int xCount = 0;
    int numCount = 0;
    
    while(getline(ss,temp,' '))//분리
        data.push_back(temp);
    
    for(string s : data)
    {
        auto it = s.find('x');
        if(it != string::npos)
        {
            if(it == 0)
                xCount++;
            else
            {
                s.replace(it, 1, "");
                xCount += stoi(s);
            }
        }
        else if(s != "+")
            numCount += stoi(s);
    }
    if(xCount != 0)
    {
        if(xCount == 1)
            answer += "x";
        else
            answer += to_string(xCount) + "x";
        
        if(numCount != 0)
            answer += " + ";
    }
    if(numCount != 0)
        answer += to_string(numCount);
    return answer;
}

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.cs
using System;

public class Solution {
    public string solution(string polynomial) {
        string answer = "";
        string[] elem = polynomial.Split(' ');
        int xCount = 0;
        int numCount = 0;
        foreach(string s in elem)
        {
            if(s.Contains('x'))
            {
                if(s[0] == 'x')
                    xCount++;
                else
                {
                    String num = s.Replace("x", "");
                    xCount += Int32.Parse(num);
                }
            }
            else if(s != "+")
                numCount += Int32.Parse(s);
        }
        if(xCount != 0)
        {
            if(xCount == 1)
                answer += "x";
            else
                answer += $"{xCount}x";

            if(numCount != 0)
                answer += " + ";
        }
        
        if(numCount != 0)
            answer += $"{numCount}";
        
        return answer;
    }
}